### DISCUSSION TITLE
access super class in __init__

### DISCUSSION THREAD

**Discussion by ghost** [hylang/hy#2448] [Q&A] [answered]
Allô,

I am working through an example for the Arcade game engine. It starts by extending the base game window class.

```
(defclass JetShooter [arcade.Window]
  (defn __init__ [self]
      (super.__init__ SCREEN_WIDTH SCREEN_HEIGHT SCREEN_TITLE)
          (setv self.enemies_list (arcade.SpriteList))
          (setv self.clouds_list (arcade.SpriteList))
          (setv self.all_sprites (arcade.SpriteList)))
```
Here is the error message from Hy; line 33 is the call to super.__init__.
```
  File "/home/claude/Programming/Python/JetAvoidsMissiles/main.hy", line 33, in __init__
    (super.__init__ SCREEN_WIDTH SCREEN_HEIGHT SCREEN_TITLE)
TypeError: descriptor '__init__' requires a 'super' object but received a 'int'
```
The Python code looks like this.
```
super().__init__(SCREEN_WIDTH, SCREEN_HEIGHT, SCREEN_TITLE)
```

I assume that the call `super()` returns a reference to the super object.

How does Hy express this Python idiom?

What is the correct way to initialize the super class?

Is there a document about translating Python to Hy?

Merci.

-- 
Claude

**Comment by Kodiologist** [comment, marked as answer]
`super().__init__(a, b, c)` in Python is written `(.__init__ (super) a b c)` in Hy. `(super.__init__ a b c)` is different because you're not actually calling `super`, just taking the `__init__` attribute of `super` itself.

> Is there a document about translating Python to Hy?

There's the manual, and there's `hy2py`, but that's about it.

**Reply by ghost** [reply]
That makes sense. Merci.
